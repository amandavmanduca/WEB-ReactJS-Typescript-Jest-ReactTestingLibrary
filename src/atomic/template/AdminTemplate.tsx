import Text from '../atom/Text'
import styles from './index.module.css';

type Props = {
    children: any;
}

export const AdminTemplate = ({children}: Props) => {
    return (
        <div className={styles.adminTemplate}>
            <div className={styles.adminTopBar}>
                <div>
                    <Text type="h2">SISTEMA ADMINISTRATIVO</Text>
                    <Text type="p">ÁREA RESTRITA</Text>
                </div>
                <div className={styles.adminAvatarBox}>
                    <img className={styles.avatarImage} src="/icons/avatar.png" alt="User" />
                    <div className={styles.userInfo}>
                        <Text type="h5">Nome do Usuário</Text>
                        <Text type="p">Administrador</Text>
                    </div>
                </div>
            </div>
            <div className={styles.adminBody}>
                <div className={styles.bodyDisplay}>
                    {children}
                </div>
            </div>
        </div>
    )
}