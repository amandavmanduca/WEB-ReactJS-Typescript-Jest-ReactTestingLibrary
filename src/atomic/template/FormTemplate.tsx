import React from 'react'
import { Form } from 'react-final-form'
import { Button } from '../atom/Button'

export const FormTemplate = ({
    formProps,
    buttonLabel,
    onCancel,
    ...props
}: any) => {
    return (
        <Form {...formProps}>
        {({
          handleSubmit,
          form,
          hasValidationErrors,
        }) => (
          <div {...props}>
            <div>
              {props.children &&
                React.cloneElement(props.children as any, { form })}
            </div>
            <div style={{ display: 'flex', gap: '10px', padding: '20px 0px', flexWrap: 'wrap' }}>
              <Button
                disabled={hasValidationErrors}
                onClick={(event: any) => handleSubmit(event)}
              >
                {buttonLabel}
              </Button>
              <Button
                onClick={onCancel}
                type={'secondary'}
              >
                  Cancelar
              </Button>
            </div>
          </div>
        )}
      </Form>
    )
}