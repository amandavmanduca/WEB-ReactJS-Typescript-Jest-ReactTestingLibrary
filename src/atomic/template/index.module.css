.adminTemplate {
    width: 100%;
    min-height: 100vh;
    background-color: #eeeeea;
}

.adminTopBar {
    display: flex;
    padding: 30px;
    background-color: white;
    min-height: 70px;
    justify-content: space-between;
    box-shadow: 0px 4px 8px 0px rgba(0, 0, 0, 0.1);
    align-items: center;
}

.adminAvatarBox {
    display: flex;
    gap: 15px;
    align-items: center;
}

.avatarImage {
    width: 40px;
    height: 40px;
    border-radius: 50%;
}

.userInfo {
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.adminBody {
    display: flex;
    width: 100%;
    height: 100%;
}

.bodyDisplay {
    width: 100%;
    height: 100%;
    padding: 30px 80px;
}


@media (max-width: 800px) {
    .bodyDisplay {
        padding: 30px 20px;
    }
}