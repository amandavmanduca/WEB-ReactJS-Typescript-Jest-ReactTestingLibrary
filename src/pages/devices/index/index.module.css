
.deviceCardBox {
    width: 100%;
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    gap: 20px;
}

@media (max-width: 1000px) {
    .deviceCardBox {
        grid-template-columns: 1fr 1fr;
    }
}

@media (max-width: 700px) {
    .deviceCardBox {
        grid-template-columns: 1fr;
    }
}

.flexBox {
    display: flex;
    flex-direction: column;
    gap: 10px;
}

.noDataFoundBox {
    width: 100%;
    margin-top: 50px;
    height: 100%;
    gap: 10px;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
}