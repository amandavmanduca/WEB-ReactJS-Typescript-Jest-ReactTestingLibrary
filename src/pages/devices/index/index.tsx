import { useEffect } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { Button } from '../../../atomic/atom/Button';
import Text from '../../../atomic/atom/Text';
import { Device } from '../../../features/common/types';
import { DeviceCard } from '../../../features/devices/components/DeviceCard';
import { useDeleteDevice } from '../../../features/devices/hooks/useDeleteDevice';
import { useListDevices } from '../../../features/devices/hooks/useListDevices';
import styles from './index.module.css';


const DevicesPage = () => {
    const { listDevices, data } = useListDevices()
    const { deleteDevice } = useDeleteDevice()
    const navigate = useNavigate();

    useEffect(() => {
        listDevices()
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])

    async function handleDelete(id: string) {
        if (id) {
            await deleteDevice(id)
            listDevices()
        }
    }

    return (
        <div className={styles.flexBox}>
            <div className={styles.flexBox}>
                <div>
                    <Button
                        onClick={() => navigate('/device/create')}
                    >
                        Criar dispositivo
                    </Button>
                </div>
                <Text type="h1">Listagem de dispositivos</Text>
            </div>
            {data?.length > 0 ? (
                <div data-testid="device-card-section" className={styles.deviceCardBox}>
                    {data?.map((device: Device) => (
                        <DeviceCard
                            key={device.id}
                            device={device}
                            deleteDevice={() => handleDelete(device.id)}
                        />
                    ))}
                </div>
            ) : (
                <div data-testid="empty-data" className={styles.noDataFoundBox}>
                    <Text type="h1">Nenhum dispositivo encontrado</Text>
                    <Link to="/device/create"><Text type="p">Comece a cadastrar clicando aqui.</Text></Link>
                    <img src="/icons/empty-folder.png" alt="Empty" />
                </div>
            )}
        </div>
    )
}

export default DevicesPage