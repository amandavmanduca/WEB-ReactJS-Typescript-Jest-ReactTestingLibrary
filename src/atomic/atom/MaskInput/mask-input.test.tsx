import { render, screen, cleanup } from '@testing-library/react';
import { MaskInput } from './index';
import renderer from 'react-test-renderer';
import { FormTemplate } from '../../template/FormTemplate';
import userEvent from '@testing-library/user-event';

afterEach(() => cleanup())

test('Should render mask input component', async () => {
    const onSubmit = () => jest.fn()
  render(
      <FormTemplate
        onCancel={jest.fn()}
        formProps={{
          onSubmit,
          initialValues: {},
        }}
    >
          <MaskInput
            validate="string"
            mask={"00000"}
            name="test"
            label="Test"
          />
      </FormTemplate>
  )
  expect(screen.getByTestId('mask-input')).toBeInTheDocument()
});

test('Should recognize mask input initial value', async () => {
  const onSubmit = () => jest.fn()
render(
    <FormTemplate
      onCancel={jest.fn()}
      formProps={{
        onSubmit,
        initialValues: {
          test: "12345"
        },
      }}
  >
        <MaskInput
          validate="string"
          mask={"00000"}
          name="test"
          label="Test"
        />
    </FormTemplate>
)
expect(screen.getByTestId('mask-input')).toHaveValue('12345')
});

test('Should recognize mask input value', async () => {
  const onSubmit = () => jest.fn()
render(
    <FormTemplate
      onCancel={jest.fn()}
      formProps={{
        onSubmit,
        initialValues: {},
      }}
  >
        <MaskInput
          validate="string"
          mask={"00000"}
          name="test"
          label="Test"
        />
    </FormTemplate>
)
userEvent.type(screen.getByTestId('mask-input'), "22222")
expect(screen.getByTestId('mask-input')).toHaveValue('22222')
});

test('Should recognize updated mask input value', async () => {
  const onSubmit = () => jest.fn()
render(
    <FormTemplate
      onCancel={jest.fn()}
      formProps={{
        onSubmit,
        initialValues: {},
      }}
  >
        <MaskInput
          validate="string"
          mask={"00000"}
          name="test"
          label="Test"
        />
    </FormTemplate>
)
userEvent.clear(screen.getByTestId('mask-input'))
userEvent.type(screen.getByTestId('mask-input'), "22222")
expect(screen.getByTestId('mask-input')).toHaveValue('22222')
});

test('Should display mask input validation error when null', async () => {
  const onSubmit = () => jest.fn()
  render(
    <FormTemplate
      onCancel={jest.fn()}
      formProps={{
        onSubmit,
        initialValues: {},
      }}
    >
      <MaskInput
        validate="string"
        mask={"00000"}
        name="test"
        label="Test"
      />
    </FormTemplate>
  )
  expect(screen.getByTestId('mask-input-validation-error')).toBeInTheDocument()
});

test('Should not display mask input validation error when filled correctly', async () => {
  const onSubmit = () => jest.fn()
  render(
    <FormTemplate
      onCancel={jest.fn()}
      formProps={{
        onSubmit,
        initialValues: {
          test: '12345'
        },
      }}
    >
      <MaskInput
        validate="string"
        mask={"00000"}
        name="test"
        label="Test"
      />
    </FormTemplate>
  )
  expect(screen.queryByTestId('mask-input-validation-error')).not.toBeInTheDocument()
});

test('Should display mask input validation error when filled incorrectly', async () => {
  const onSubmit = () => jest.fn()
  render(
    <FormTemplate
      onCancel={jest.fn()}
      formProps={{
        onSubmit,
        initialValues: {},
      }}
    >
      <MaskInput
        validate="string|length:5"
        mask={"00000"}
        name="test"
        label="Test"
      />
    </FormTemplate>
  )
  userEvent.type(screen.getByTestId('mask-input'), "123")
  expect(screen.getByTestId('mask-input-validation-error')).toBeInTheDocument()
});

test('Should only be able do digit numbers considering mask input options', async () => {
  const onSubmit = () => jest.fn()
  render(
    <FormTemplate
      onCancel={jest.fn()}
      formProps={{
        onSubmit,
        initialValues: {},
      }}
    >
      <MaskInput
        validate="string"
        mask={"00000"}
        name="test"
        label="Test"
      />
    </FormTemplate>
  )
  userEvent.type(screen.getByTestId('mask-input'), "ABC")
  expect(screen.getByTestId('mask-input')).toHaveValue('')
});

test('Should only accept digit conforming mask input options', async () => {
  const onSubmit = () => jest.fn()
  render(
    <FormTemplate
      onCancel={jest.fn()}
      formProps={{
        onSubmit,
        initialValues: {},
      }}
    >
      <MaskInput
        validate="string"
        mask={"00000"}
        name="test"
        label="Test"
      />
    </FormTemplate>
  )
  userEvent.type(screen.getByTestId('mask-input'), "ABC123")
  expect(screen.getByTestId('mask-input')).toHaveValue('123')
});

test('Should not be able to digit more than expected by mask validation', async () => {
  const onSubmit = () => jest.fn()
  render(
    <FormTemplate
      onCancel={jest.fn()}
      formProps={{
        onSubmit,
        initialValues: {},
      }}
    >
      <MaskInput
        validate="string|length:3"
        mask={"000"}
        name="test"
        label="Test"
      />
    </FormTemplate>
  )
  userEvent.type(screen.getByTestId('mask-input'), "12345")
  expect(screen.getByTestId('mask-input')).toHaveValue('123')
});

test('Mask input component should matches snapshot', async () => {
    const onSubmit = () => jest.fn()
    const tree = renderer.create(<FormTemplate
        onCancel={jest.fn()}
        formProps={{
          onSubmit,
          initialValues: {},
        }}
    >
          <MaskInput
            validate="string"
            mask={"00000"}
            name="test"
            label="Test"
          />
      </FormTemplate>).toJSON()
    expect(tree).toMatchSnapshot()
})
