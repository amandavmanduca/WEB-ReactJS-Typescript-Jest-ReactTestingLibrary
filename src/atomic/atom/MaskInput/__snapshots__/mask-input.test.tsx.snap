// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mask input component should matches snapshot 1`] = `
<div>
  <div>
    <div>
      <label
        htmlFor="test"
      >
        <p
          className="inputLabel"
        >
          Test
        </p>
      </label>
      <input
        className="formInput"
        data-testid="mask-input"
        id="test"
        name="test"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
      />
      <div />
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "gap": "10px",
        "padding": "20px 0px",
      }
    }
  >
    <button
      className="button"
      data-testid="button"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#fc8eac",
        }
      }
    >
      <p
        className="buttonText"
      />
    </button>
    <button
      className="button"
      data-testid="button"
      disabled={false}
      onClick={[MockFunction]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <p
        className="secondaryButtonText"
      >
        Cancelar
      </p>
    </button>
  </div>
</div>
`;
